imap_processing/spice/kernels.py

```diff
@@ -156,3 +159,215 @@ def wrapper_ensure_spice(*args: Any, **kwargs: Any) -> Any:
         return _decorator(f_py)
     else:
         return _decorator
+
+
+@ensure_spice
+def create_pointing_frame(pointing_frame_dir: Optional[Path] = None) -> Path:
+    """
+    Create the pointing frame.
+
+    Parameters
+    ----------
+    pointing_frame_dir : Path
+        Directory of where pointing frame will be saved.
+
+    Returns
+    -------
+    path_to_pointing_frame : Path
+        Path to pointing frame.
+    """
+    ck_kernel, _, _, _ = spice.kdata(0, "ck")
+    directory = Path(ck_kernel).parent
+
+    # Get timerange for the pointing frame kernel.
+    et_start, et_end, et_times = _get_et_times(ck_kernel)
+    # Create a rotation matrix
+    rotation_matrix = _create_rotation_matrix(et_times)
+
+    # Convert the rotation matrix to a quaternion.
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.m2q
+    q_avg = spice.m2q(rotation_matrix)
+
+    # TODO: come up with naming convention.
+    if pointing_frame_dir is None:
+        path_to_pointing_frame = directory / "imap_dps.bc"
+    else:
+        path_to_pointing_frame = pointing_frame_dir / "imap_dps.bc"
```

Comment:
Do we need this to be optional / configurable, or can we say something like "write the frame kernel to the same location as the input CK with the name ..."

I'd suggest changing to the full path to be passed in if we do want to go that route rather than a directory since that is more explicit for end users (bringing your `directory` variable inline here directly too)

```suggestion
    # TODO: come up with naming convention.
    if pointing_frame_path is None:
        pointing_frame_path = Path(ck_kernel).parent / "imap_dps.bc"
```